### ISSUE TITLE
Connection issues

### ISSUE THREAD

**Issue by ispyisail** [Tertiush/Alarmin-Android-App#30]
Hi

Does anybody else have connection problems? I've never got the WAN connection to work (IP or dynamic address).

The LAN connection works every second time (Mostly ). The first connection always fails but if I click back then have a second go it mostly works?

**Comment by Tertiush** [comment]
Hi

I've also seen this at random times. I think this is largely due to GSM
networks not being able to fulfill/complete a connection quick enough
before the app times out. I will have a look into it, perhaps by sending
multiple initial requests, using the initial ones to prompt the link into
action. I think this might do the job.

Regarding the WAN connections, feel free to send me a log file to have a
look. Just follow the instructions on top of the log screen (see the app's
menu). I have not had anyone else recently with WAN issues.

**Comment by Christofinity** [comment]
I'm also having issues connecting via WAN. When I'm connecting through my internal LAN, it works. But when I switch to 4G Mobile data, then error comes up. I've got a Static IP address and have port 80 as webport and 10000 as software port. Both of those ports are open as I checked with http://canyouseeme.org. 

I have sent you the log via the app.

**Comment by Tertiush** [comment]
I got the log thanks. Please see if you can connect to the alarm using your phone's web browser - and take note of any delays you experience. If the browser works fine (and quick) please send another log with the level set to verbose (look in settings). I'm not sure what the issue could be yet, hopefully this test will help us...

**Comment by Christofinity** [comment]
I made an account with paradoxmyhome and my IP150 device is connected to it.

I'm using 4G mobile data and put my WAN Static IP address in my phones web browser and I get to the IP150 module login screen. I can login to the module fine. Is that what you wanted me to do?

After I've logged into my module via phone web browser. Did you want me to open the Alarmin app and go into the log and send to you again? I'm not quite following...

**Comment by Tertiush** [comment]
Ok - yes I just wanted to confirm that you can get to the ip module through your 4G link and phone's browser. Seeing as that works, please send another log, but first read the instructions on the top of the logs page, it needs to be set to verbose mode.

**Comment by Christofinity** [comment]
Did you want me to try and connect first, then after the error comes up go into the log and send?

**Comment by Tertiush** [comment]
Yes, if you read the instruction it will ask to repeat the thing you are having an issue with before send ing the logs. So first change the logs to verbose, then try to connect, then send the logs

**Comment by ispyisail** [comment]
After some thought I'm 99 percent sure the problem is blocked ports. We
need to forward the correct port from our router to IP module?

**Comment by Christofinity** [comment]
In my case, it's not blocked ports. As both port 80 and 10000 are opened.

But we have now seemed to have found the problem with my case. Alarmin seems to not like certain ROMs. As I'm using a Xiaomi phone running MIUI and this issue is happening. But Tertiush asked me to try with another phone and it worked. The other phone I tried with and worked was the Moto G.

**Comment by Christofinity** [comment]
If there is another Alarmin user here who has got a Xiaomi phone and/or running MIUI can post here whether or not Alarmin is working fine over an external network or not, it would be much appreciated. Just incase it's actually not MIUI that's the issue.

**Comment by sergiobaron** [comment]
Hi.
I have a problem with WAN connection and ony in 4g. All the time says that the connection panel are inestable.
It works well with WIFI connection.

**Comment by ispyisail** [comment]
The problem will most likely be with ports/port forwarding

local/wifi is easier to set up

**Comment by sergiobaron** [comment]
Thanks for your info, but the connection works well with wifi connection only to access to internet, in a different lan than the ip150 module is connected, so they use the same port fowarding than 4g to access to the ip150 module.

**Comment by ispyisail** [comment]
Mine works perfect. We need to see a screen shot of your setup. It's most
likely a setup issue.

**Comment by sergiobaron** [comment]
The connection works well if I connect with any wifi, even in a different place and lan where the ip150 module is connected
